File: Js_functions/recursion/recurse.py
import sys

def count_down(num):
    #The function recurses and makes a count down till 0

    print(num)
    new_number = num -1

    if new_number > 0 :
        count_down(new_number)

if len(sys.argv) < 2 :
    print("Sorry no input. Program shutting down")
    sys.exit(1)

try:
    number = int(sys.argv[1])
except ValueError:
    print("Invalid Number")
    sys.exit(1)


count_down(number)

File: Js_functions/recursion/recurse.js
function countDown(number){
    // function that recurses as it counts down
    console.log(number);
    let new_number = number - 1;

    if(new_number > 0){
        countDown(new_number)
    }
}

if(process.argv.length < 3){
    console.log("Please enter a number to count down from");
    process.exit(1)
}

const input = process.argv[2];
const number = parseInt(input);

if(isNaN(number)) {
    console.log("Invalid number!");
    process.exit(1)
}

countDown(number);